### PR TITLE
For #18521 - Added scroll to current tab

### PR DESCRIPTION
**Pull request by codrut-topliceanu**
-> https://github.com/mozilla-mobile/fenix/issues/18521
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

https://user-images.githubusercontent.com/60002907/114707405-da389900-9d32-11eb-9b00-2f5bb3eea73f.mp4

### CODE REVIEW

**Review by jonalmeida — APPROVED** [review]
Some small nits on design, but looks good otherwise! Let's land this with some tests and we can address the issues in a follow-up.

File: app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt [hunk 1]
```diff
@@ -8,45 +8,74 @@ import android.content.Context
...
     private val browserInteractor: BrowserTrayInteractor,
     private val syncedTabsInteractor: SyncedTabsView.Listener,
-    private val interactor: TabsTrayInteractor
+    private val interactor: TabsTrayInteractor,
+    private val browserStore: BrowserStore
 ) : RecyclerView.Adapter<AbstractTrayViewHolder>() {
 
-    private val normalAdapter by lazy { BrowserTabsAdapter(context, browserInteractor, store) }
-    private val privateAdapter by lazy { BrowserTabsAdapter(context, browserInteractor, store) }
+    private val normalAdapter by lazy {
```
[the remark this answers is not in the document]
Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt:33]: Sorry about that. I'm usually very careful to only auto-format code I've written myself.

File: app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt [hunk 2]
```diff
@@ -8,45 +8,74 @@ import android.content.Context
...
+    }
+    private val privateAdapter by lazy {
+        BrowserTabsAdapter(
+            context,
+            browserInteractor,
+            store
+        )
+    }
     private val syncedTabsAdapter by lazy { SyncedTabsAdapter(syncedTabsInteractor) }
+    private lateinit var viewHolder: AbstractTrayViewHolder
```
[the remark this answers is not in the document]
Comment (codrut-topliceanu) [reply app/src/main/java/org/mozilla/fenix/tabstray/TrayPagerAdapter.kt:44]: The original code I wrote was using it. There's no need for it now. Sorry bout that 😅 .

**Comment by codrut-topliceanu** [comment]
As discussed tests will be added in a follow-up.

**Pull request merged**